cmd/action/migrate.go

```diff
@@ -0,0 +1,179 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"text/template"
+	"time"
+
+	"ariga.io/atlas/sql/migrate"
+	"ariga.io/atlas/sql/schema"
+	"github.com/spf13/cobra"
+)
+
+const (
+	migrateFlagDir        = "dir"
+	migrateFlagSchema     = "schema"
+	migrateFlagForce      = "force"
+	migrateDiffFlagDevURL = "dev-url"
+	migrateDiffFlagTo     = "to"
+)
+
+var (
+	// MigrateFlags are the flags used in MigrateCmd (and sub-commands).
+	MigrateFlags struct {
+		DirURL  string
+		DevURL  string
+		ToURL   string
+		Schemas []string
+		Force   bool
+	}
+	// MigrateCmd represents the migrate command. It wraps several other sub-commands.
+	MigrateCmd = &cobra.Command{
+		Use:   "migrate",
+		Short: "Manage versioned migration files",
+		Long:  "`atlas migrate`" + ` wraps several sub-commands for migration management.`,
+		PersistentPreRun: func(*cobra.Command, []string) {
+			// Migrate commands will not run on a broken migration directory, unless the force flag is given.
+			if !MigrateFlags.Force {
+				dir, err := dir()
+				cobra.CheckErr(err)
+				cobra.CheckErr(migrate.Validate(dir)) // TODO(masseelch): tell the user what's wrong
+			}
+		},
+	}
+	// MigrateDiffCmd represents the migrate diff command.
+	MigrateDiffCmd = &cobra.Command{
+		Use:   "diff",
+		Short: "Compute the diff between the migration directory and a connected database and create a new migration file.",
+		Long: "`atlas migrate diff`" + ` uses the dev-database to re-run all migration files in the migration
+directory and compares it to a given desired state and create a new migration file containing SQL statements to migrate 
+the migration directory state to the desired schema. The desired state can be another connected database or an HCL file.`,
+		Example: `  atlas migrate diff --dev-url mysql://user:pass@localhost:3306/dev --to mysql://user:pass@localhost:3306/dbname
+  atlas migrate diff --dev-url mysql://user:pass@localhost:3306/dev --to file://atlas.hcl`,
+		Args: cobra.MaximumNArgs(1),
+		Run:  CmdMigrateDiffRun,
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(MigrateCmd)
+	MigrateCmd.AddCommand(MigrateDiffCmd)
+	MigrateCmd.PersistentFlags().StringVarP(&MigrateFlags.DirURL, migrateFlagDir, "", "file://migrations", "select migration directory using DSN format")
+	MigrateCmd.PersistentFlags().StringSliceVarP(&MigrateFlags.Schemas, migrateFlagSchema, "", nil, "set schema names")
```

Comment:
Move it close to the `ToURL`.

Reply:
`--schemas` is a global flag. `-to` is not :-(